sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/CanCreateSparkDataFrame.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2020 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.dataobject
+
+import io.smartdatalake.workflow.dataframe.GenericDataFrame
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.workflow.{ActionPipelineContext, DataFrameSubFeed}
+import io.smartdatalake.workflow.dataframe.spark.{SparkDataFrame, SparkSubFeed}
+import org.apache.spark.sql.{DataFrame, DataFrameReader, DataFrameWriter}
+
+import scala.reflect.runtime.universe.{Type, typeOf}
+
+private[smartdatalake] trait CanCreateSparkDataFrame { this: DataObject with CanCreateDataFrame => // CanCreateDataFrame is not extended because a DataObject can implement multiple CanCreate*DataFrame traits.
```

Comment:
I don't get why it's a problem to have a DataObject extend multiple CanCreate*DataFrame traits. Is it because of conflicting implementations of methods?
I tried to reconstruct it with a simple code snippet, but did not get what's the problem:

    trait Base {
    def blub: Int
    }
    
    trait SubA extends Base{
    override def blub = 1
    }
    
    trait SubB extends Base {
    override def blub: Int = 2
    }
    
    case class SubSub() extends SubB with SubA {
    
    }
    val subsub1 = SubSub()
    println(subsub1.blub)

Reply:
You're right. There is no problem with inheritance. It was already improved in CanWriteSparkDataFrame, i changed this also in CanReadSparkDataFrame.